File: C++/SelectionSort/src/Author.cpp
#include "Author.h"

Author::Author(string _firstName, string _lastName) {
	firstName = _firstName;
	lastName = _lastName;
}

int Author::compareTo(Author _author) {

	int compare = 0;
	if(this->lastName > _author.getLastName()) {
		compare = 1;
	} else if(this->lastName < _author.getLastName()) {
		compare = -1;
	}

	return compare;
}

File: Python/SelectionSort/src/Author.py

class Author:
    def __init__(self, firstName, lastName):
        self.__firstName = firstName
        self.__lastName = lastName

    def compareTo(self, author):
        compare = 0
        if self.__lastName > author.getLastName():
            compare = 1
        elif self.__lastName < author.getLastName():
            compare = -1
        return compare

    def getFirstName(self):
        return self.__firstName

    def getLastName(self):
        return self.__lastName
